tls/s2n_server_cert.c

```diff
@@ -23,8 +23,42 @@
 
 #include "utils/s2n_safety.h"
 
+static int inline is_cert_supported(struct s2n_connection *conn, s2n_cert_type actual_cert_type)
+{
+    if (conn->actual_protocol_version == S2N_TLS13) {
+        /* in TLS 1.3, the ciphersuite's auth_method
+         * S2N_AUTHENTICATION_METHOD_TLS13 allows RSA and ECDSA certs */
+        switch (actual_cert_type) {
+        case S2N_CERT_TYPE_RSA_SIGN:
+        case S2N_CERT_TYPE_ECDSA_SIGN:
+            break;
+        default:
+            S2N_ERROR(S2N_ERR_CERT_TYPE_UNSUPPORTED);
+        }
+    } else {
+        s2n_authentication_method expected_auth_method = conn->secure.cipher_suite->auth_method;
+        switch (actual_cert_type) {
+        case S2N_CERT_TYPE_RSA_SIGN:
+            S2N_ERROR_IF(expected_auth_method != S2N_AUTHENTICATION_RSA, S2N_ERR_CERT_TYPE_UNSUPPORTED);
```

Comment:
TLS1.2 supports rsa_sign as well: https://tools.ietf.org/html/rfc5246#section-7.4.4

Reply:
it is supported, but would be configured specifically with cipher_suite->auth_method